borg/repository.py

```diff
@@ -83,6 +85,9 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         if exc_type is not None:
             self.rollback()
+            if hasattr(exc_val, 'errno'):
+                if exc_val.errno in [errno.EIO, errno.ENXIO]:
+                    return
         self.close()
```

Comment:
> If what you mean by _I don't see the difference to your earlier approach?_ ...

I mean that it performs the same error handling as before just written down a different way.

> It's true that the exceptions that "travel by" this method could've been raised pretty much anywhere. Do you think errno.EIO and errno.ENXIO could have been caused by other than repository/cache related OSError?

Exactly. One can't distinguish the two cases (repository/cache-caused or input-caused) by looking at error codes.